<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="a1" doc="Robot type: [sirius_mini, sirius_mid, sirius_belet, sirius_plus]"/>
    <arg name="cheater" default="false"/>
    <arg name="use_js" default="true"/>

    <!-- make the files into global parameters -->
    <param name="urdfFile" value="/tmp/legged_control/$(arg robot_type).urdf"/>
    <param name="taskFile" value="$(find legged_controllers)/config/$(arg robot_type)/task.info"/>
    <param name="referenceFile" value="$(find legged_controllers)/config/$(arg robot_type)/reference.info"/>
    <param name="gaitCommandFile" value="$(find legged_controllers)/config/$(arg robot_type)/gait.info"/>

    <rosparam file="$(find legged_controllers)/config/controllers.yaml" command="load"/>

    <node if="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/legged_controller
          controllers/legged_cheater_controller
"/>

    <node unless="$(arg cheater)" name="controller_manager" pkg="controller_manager" type="spawner" respawn="false"
          output="screen" args="
          controllers/joint_state_controller
          controllers/legged_controller
"/>

    <!-- <node pkg="legged_controllers" type="legged_target_trajectories_publisher" name="legged_robot_target"
          output="screen"/> -->
   <node pkg="ocs2_legged_robot_ros" type="legged_robot_target" name="legged_robot_target"
          output="screen" launch-prefix="gnome-terminal --"/>     

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
          output="screen" launch-prefix="gnome-terminal --tab -t gait --  "/>

    <group if="$(arg use_js)">
            <include file="$(find legged_controllers)/launch/joy_teleop.launch"/>
    </group>

</launch>
